Stripe/PublicHeaders/STPApplePayContext.h

```diff
@@ -0,0 +1,132 @@
+ 
+ You may also implement the optional delegate methods to handle shipping methods and shipping address changes e.g. to verify you can ship to the address, or update the payment amount.
+ */
+@protocol STPApplePayContextDelegate <NSObject>
+
+/**
+ Called after the customer has authorized Apple Pay.  Implement this method to call the completion block with the client secret of a PaymentIntent representing the payment.
+ 
+ @param paymentMethodID       The identifier of the PaymentMethod that represents the customer's Apple Pay payment method.
+ If you create the PaymentIntent with confirmation_method=manual, pass this as the payment_method and confirm=true. Otherwise, you can ignore this parameter.
+ @param completion                  Call this with the PaymentIntent's client secret, or the error that occurred creating the PaymentIntent.
+ */
+- (void)applePayContext:(STPApplePayContext *)context
+ didCreatePaymentMethod:(NSString *)paymentMethodID
+             completion:(STPIntentClientSecretCompletionBlock)completion;
+
+/**
+ Called after the Apple Pay sheet is dismissed with the result of the payment.
+ 
+ Your implementation could stop a spinner and display a receipt view or error to the customer, for example.
+ 
+ @param status The status of the payment
+ @param error The error that occurred, if any.
+ */
+- (void)applePayContext:(STPApplePayContext *)context
+  didCompleteWithStatus:(STPPaymentStatus)status
+                  error:(nullable NSError *)error;
+
+@optional
+
+/**
+  Called when the user selects a new shipping method.  The delegate should determine
+  shipping costs based on the shipping method and either the shipping address supplied in the original
+  PKPaymentRequest or the address fragment provided by the last call to paymentAuthorizationViewController:
+  didSelectShippingContact:completion:.
+ 
+  You must invoke the completion block with an updated array of PKPaymentSummaryItem objects.
+ */
+- (void)applePayContext:(STPApplePayContext *)context
+didSelectShippingMethod:(PKShippingMethod *)shippingMethod
+                handler:(void (^)(PKPaymentRequestShippingMethodUpdate *update))completion API_AVAILABLE(ios(11.0), watchos(4.0));
+
+/**
+ Called when the user has selected a new shipping address.  You should inspect the
+ address and must invoke the completion block with an updated array of PKPaymentSummaryItem objects.
+ */
+- (void)applePayContext:(STPApplePayContext *)context
+didSelectShippingContact:(PKContact *)contact
+                handler:(void (^)(PKPaymentRequestShippingContactUpdate *update))completion API_AVAILABLE(ios(11.0), watchos(4.0));
+
+/**
+ A pre-iOS 11 version of paymentAuthorizationViewController:didSelectShippingContact:handler:
+ */
+- (void)applePayContext:(STPApplePayContext *)context
+didSelectShippingContact:(PKContact *)contact
+             completion:(void (^)(PKPaymentAuthorizationStatus status, NSArray<PKShippingMethod *> *shippingMethods,
```

Comment:
we're inconsistent on completion/handler in these delegate methods. I think the argument name and label should match, i.e. `completion:(type)completion` or `handler:(type)handler`. I'd vote for `completion` but am fine with either

Reply:
weird. I'd vote for changing to `handler:(type)handler` and `completion:(type)completion` but won't block